core/templates/dev/head/editor/StateResponses.js

```diff
@@ -463,12 +463,15 @@ oppia.controller('StateResponses', [
                 $scope.trainingDataFeedback = feedback;
                 $scope.trainingDataOutcomeDest = dest;
 
-                if (classificationResult.rule_spec_string !== DEFAULT_RULE_NAME &&
-                    classificationResult.rule_spec_string !== FUZZY_RULE_TYPE) {
+                var answerGroupIndex = classificationResult.answerGroupIndex;
+                var ruleSpecIndex = classificationResult.ruleSpecIndex;
+                if (answerGroupIndex !== _state.interaction.answer_groups.length &&
+                    _state.interaction.answer_groups[answerGroupIndex].rules_specs[
```

Comment:
Good catch! I don't think there are any tests for this part, but I think adding tests for this is outside the scope of this PR?